### ISSUE TITLE
Add learning rate finder

### ISSUE THREAD

**Issue by pokey** [chaosparrot/parrot.py#30]
As a related quick note to self: In addition to implementing some sort of learning rate scheduler, we should also add a learning rate finder. Lightning seems to ship with one: https://pytorch-lightning.readthedocs.io/en/1.4.5/advanced/lr_finder.html

_Originally posted by @ym-han in https://github.com/chaosparrot/parrot.py/issues/18#issuecomment-1272740997_
